tsdb/querier.go

```diff
@@ -550,65 +555,70 @@ func (p *populateWithDelGenericSeriesIterator) toChunkSeriesIterator() chunks.It
 // populateWithDelSeriesIterator allows to iterate over samples for the single series.
 type populateWithDelSeriesIterator struct {
 	*populateWithDelGenericSeriesIterator
-
-	curr chunkenc.Iterator
 }
 
 func (p *populateWithDelSeriesIterator) Next() bool {
-	if p.curr != nil && p.curr.Next() {
-		return true
+	if p.currChkIter == nil {
```

Comment:
Optimization num 2: Simplifying and making sure we can reuse chunkIterator and bufDelIterator correctly.